relay-kafka/src/config.rs

```diff
@@ -80,93 +80,62 @@ impl KafkaTopic {
-    /// Topic name for all other kinds of metrics. Defaults to the assignment of `metrics`.
-    #[serde(alias = "metrics_transactions", alias = "ingest-performance-metrics")]
-    pub metrics_generic: TopicAssignment,
-    /// Stacktrace topic name
-    pub profiles: TopicAssignment,
-    /// Replay Events topic name.
-    #[serde(alias = "ingest-replay-events")]
-    pub replay_events: TopicAssignment,
-    /// Recordings topic name.
-    #[serde(alias = "ingest-replay-recordings")]
-    pub replay_recordings: TopicAssignment,
-    /// Monitor check-ins.
-    #[serde(alias = "ingest-monitors")]
-    pub monitors: TopicAssignment,
-    /// Standalone spans without a transaction.
-    #[serde(alias = "snuba-spans")]
-    pub spans: TopicAssignment,
-    /// Summary for metrics collected during a span.
-    #[serde(alias = "snuba-metrics-summaries")]
-    pub metrics_summaries: TopicAssignment,
-    /// COGS measurements.
-    #[serde(alias = "shared-resources-usage")]
-    pub cogs: TopicAssignment,
-}
+macro_rules! define_topic_assignments {
+    ($struct_name:ident {
+        $($field_name:ident : ($kafka_topic:path, $default_topic: literal)),* $(,)? }) => {
 
-impl TopicAssignments {
-    /// Get a topic assignment by [`KafkaTopic`] value
-    #[must_use]
-    pub fn get(&self, kafka_topic: KafkaTopic) -> &TopicAssignment {
-        match kafka_topic {
-            KafkaTopic::Attachments => &self.attachments,
-            KafkaTopic::Events => &self.events,
-            KafkaTopic::Transactions => &self.transactions,
-            KafkaTopic::Outcomes => &self.outcomes,
-            KafkaTopic::OutcomesBilling => self.outcomes_billing.as_ref().unwrap_or(&self.outcomes),
-            KafkaTopic::MetricsSessions => &self.metrics_sessions,
-            KafkaTopic::MetricsGeneric => &self.metrics_generic,
-            KafkaTopic::Profiles => &self.profiles,
-            KafkaTopic::ReplayEvents => &self.replay_events,
-            KafkaTopic::ReplayRecordings => &self.replay_recordings,
-            KafkaTopic::Monitors => &self.monitors,
-            KafkaTopic::Spans => &self.spans,
-            KafkaTopic::MetricsSummaries => &self.metrics_summaries,
-            KafkaTopic::Cogs => &self.cogs,
+        /// Configuration for topics.
+        #[derive(Serialize, Deserialize, Debug)]
+        #[serde(default)]
+        pub struct TopicAssignments {
+            $(
+                /// Topic name for $default_topic
+                #[serde(alias = $default_topic)]
+                pub $field_name: TopicAssignment,
+            )*
         }
-    }
-}
 
-impl Default for TopicAssignments {
-    fn default() -> Self {
-        Self {
-            events: "ingest-events".to_owned().into(),
-            attachments: "ingest-attachments".to_owned().into(),
-            transactions: "ingest-transactions".to_owned().into(),
-            outcomes: "outcomes".to_owned().into(),
-            outcomes_billing: None,
```

Comment:
How are we handling this special case? I assume this is the reason why integration tests are failing, see

https://github.com/getsentry/relay/blob/94cb3df759108f483b7cf3c2f41e0c6262eefb06/relay-kafka/src/config.rs#L138

Reply:
I'm not handling it at all. This is no longer handled as a special case in any our other services (see https://github.com/getsentry/sentry/blob/d414afaab6fcd22e8609d0a885e25cad0752b83b/src/sentry/conf/server.py#L3477 and https://github.com/getsentry/sentry-kafka-schemas/blob/main/topics/outcomes-billing.yaml for example). If someone reaalllly wanted to share the topic technically still they could accomplish it via overriding `outcomes-billing` back to `outcomes` in relay config (but that's not really something we'd do anymore). It only ever existed for dev and dev was updated to just match prod on this recently.

Reply:
Does it break self hosted?

Reply:
It doesn't. Self hosted runs a billing consumer too. https://github.com/getsentry/self-hosted/blob/c4ae491929170c075bea608941b227a8db81eec2/docker-compose.yml#L268-L270